src/plugins/files/public/components/upload_file/index.tsx

```diff
@@ -6,17 +6,24 @@
  * Side Public License, v 1.
  */
 
-import React, { lazy, Suspense } from 'react';
+import React, { lazy, Suspense, ReactNode } from 'react';
 import { EuiLoadingSpinner } from '@elastic/eui';
 import type { Props } from './upload_file';
 
 export type { DoneNotification } from './upload_state';
-export type { Props as UploadFileProps };
+
+export type UploadFileProps = Props & {
+  /**
+   * A custom fallback for when component is lazy loading,
+   * If not provided, <EuiLoadingSpinner /> is used
+   */
+  lazyLoadFallback?: ReactNode;
```

Comment:
I also added a way to customize this suspense fallback. I found this useful in image embeddable for avoiding layout shifts